linux/shim/init.go

```diff
@@ -70,11 +73,14 @@ func newInitProcess(context context.Context, path string, r *shimapi.CreateReque
 		IO:            io,
 		NoPivot:       r.NoPivot,
 	}
+	sigchld := make(chan os.Signal, 2048)
+	// main.go calls Notify as well, but it does not hurt
+	signal.Notify(sigchld, syscall.SIGCHLD)
```

Comment:
This would work for the init process but not for execs (which have a similar code).

For execs, you need a way to know that the child was the one for the new exec.